Update Gradle snippets

ciscoo opened the pull request:
<!---
Thanks for helping to make Cucumber better! 💖

You can feel free to open a "Draft" status pull request if you're not ready for feedback yet.

Don't worry about getting everything perfect! We're here to help and will coach you through
to getting your pull request ready to merge.

The prompts below are for guidance to help you describe your change in a way that is most 
likely to make sense to other people when they are reviewing it. Still, it's just a guide, 
so feel free to delete anything that doesn't feel appropriate, and add anything additional 
that seems like it would provide useful context. 👏🏻
-->

### 🤔 What's changed?

<!-- Describe your changes in detail -->

Updated Gradle snippets to use the correct configuration when declaring dependencies. Originally, in some places, the `compile` configuration was used. This was incorrect because generally, test dependencies should not be on the main application classpath and the `compile` configuration was removed anyway in later/modern versions of Gradle.

It does make sense to include Cucumber if one was building a testing library of some sort, but then that would be declared in the `api` configuration.

### ⚡️ What's your motivation? 

<!-- 
What motivated you to propose this change? Does it fix a bug? Add a new feature?
If it fixes an open issue, you can link to the issue here, e.g. "Fixes #99"
-->

The Gradle snippets were outdated using long removed configurations (`compile`/`testCompile`).

Additionally, there is now two DSLs for Gradle: Groovy and Kotlin. So snippets are shown for Kotlin, but since they are simple to begin with, it also works for the Groovy DSL.

### 🏷️ What kind of change is this?

<!--- Delete any options that are not relevant -->

- :book: Documentation (improvements without changing code)

### ♻️ Anything particular you want feedback on?

<!-- 
Is there anything in this change you're unsure about, or would 
particularly like reviewers to give you feedback on?
-->

### 📋 Checklist:

<!--- 
This is to help you remember all the little things we often forget to do!

Feel free to delete any tasks that are not relevant, or add new ones.
-->

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://github.com/cucumber/.github/tree/main?tab=coc-ov-file)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

mattwynne commented:
Thanks for a great post @tooky!

On Mon, Dec 21, 2015 at 9:36 AM Theo <notifications@github.com> wrote:

> Merged #97 <https://github.com/cucumber/website/pull/97>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cucumber/website/pull/97#event-497232693>.
>

ciscoo committed:
Update Gradle snippets

mpkorstanje reviewed (changes requested):
Not reviewed in detail yet. 

For the sake of maintenance, I would prefer a single DSL, perferably Kotlin. Is it currently feasible to drop the Groovy DSL?

ciscoo commented:
The syntax for Kotlin would work for both DSLs since these sinppets are simple enough, so that's perfectably feasible.

ciscoo committed:
Simplify Gradle snippets

Default to the Kotlin DSL since the snippets are simple enough they work for both Groovy and Kotlin DSLs.

mpkorstanje commented:
Cheers! Much appreciated!
